src/main/java/uk/ac/sanger/sccp/stan/service/WorkProgressServiceImp.java

```diff
@@ -80,26 +93,47 @@ public List<WorkProgress> getProgress(String workNumber, List<String> workTypeNa
             if (statuses!=null && !statuses.contains(singleWork.getStatus())) {
                 return List.of();
             }
+            if (programs!=null && !programs.contains(singleWork.getProgram())) {
+                return List.of();
+            }
             return List.of(getProgressForWork(singleWork, opTypeFilter, stainTypeFilter, labwareTypeFilter,
                     releaseLabwareTypeFilter, labwareIdToLabware, labwareTypeToStainMap));
         }
+
         if (workTypes!=null) {
             List<Work> works = workRepo.findAllByWorkTypeIn(workTypes);
             Stream<Work> workStream = works.stream();
             if (statuses!=null) {
                 workStream = workStream.filter(work -> statuses.contains(work.getStatus()));
             }
+            if (programs!=null) {
+                workStream = workStream.filter(work -> programs.contains(work.getProgram()));
+            }
             return workStream
                     .map(work -> getProgressForWork(work, opTypeFilter, stainTypeFilter, labwareTypeFilter,
                             releaseLabwareTypeFilter, labwareIdToLabware, labwareTypeToStainMap))
                     .collect(toList());
         }
+
+        if (programs!=null) {
+            List<Work> works = workRepo.findAllByProgramIn(programs);
+            Stream<Work> workStream = works.stream();
+            if (statuses!=null) {
+                workStream = workStream.filter(work -> statuses.contains(work.getStatus()));
+            }
```

Comment:
Same question as above. Also, if that's the case why are not checking against all valid work types here?

Reply:
At this point in the code, we know that `workTypes` is null